### ISSUE TITLE
Bad credential error with 0.10 release

### ISSUE THREAD

**Issue by ywei2017** [bosh-prometheus/cf_exporter#50]
After upgrading to Prometheus 26.0.1, we observed that in some of our environments (others are fine), cf_exporter keeps on failing, with the following message:
```
==> /var/vcap/sys/log/cf_exporter/cf_exporter.stderr.log <==
time="2019-11-28T14:25:56Z" level=error msg="Error while listing service bindings: Error requesting service bindings: Get https://api.system.$TLD/v2/service_bindings?: oauth2: cannot fetch token: 401 Unauthorized\nResponse: {\"error\":\"unauthorized\",\"error_description\":\"Bad credentials\"}" source="service_bindings_collector.go:142"
time="2019-11-28T14:25:56Z" level=error msg="Error while listing service planss: Error requesting service plans: Get https://api.system.$TLD/v2/service_plans?: oauth2: cannot fetch token: 401 Unauthorized\nResponse: {\"error\":\"unauthorized\",\"error_description\":\"Bad credentials\"}" source="service_plans_collector.go:142"
time="2019-11-28T14:25:56Z" level=error msg="Error while listing organization: Error requesting orgs: Get https://api.system.$TLD/v2/organizations?: oauth2: cannot fetch token: 401 Unauthorized\nResponse: {\"error\":\"unauthorized\",\"error_description\":\"Bad credentials\"}" source="applications_collector.go:208"
```
After trying many different ways (recreating the UAA account, removing possible unfriendly characters in the secrets...), I decided to try an older release. I found both 0.9 and 0.8 works fine, so it seems there is bug introduced in 0.10, that causes the error in some situations.

Let me if you want me to run any diagnostics.

Thanks
Yansheng

**Comment by xyloman** [comment]
I was able to reproduce this consistently with a client generated in the following way:

```
client_secret=$(openssl rand -base64 14)
```

When openssl generates random values it appears to almost always include an [equal sign](https://security.stackexchange.com/questions/81976/is-this-a-secure-way-to-generate-passwords-at-the-command-line/81995)  The solution was to take what is in the [stackexchange conversation](https://security.stackexchange.com/questions/81976/is-this-a-secure-way-to-generate-passwords-at-the-command-line/81995) and do the following:

```
client_secret=$(openssl rand -base64 32 | tr -d /=+ | cut -c -30)
```

This would ensure that the equal sign was not generated.  Is there something that could be updated in the cf_exporter to handle a secret with an equal sign?  I have tested locally with the binary and if the secret contains an equal sign it always fails to call the cf apis.
